src/Cardano/WalletLayer.hs

```diff
@@ -0,0 +1,24 @@
+-- |
+-- Copyright: © 2018-2019 IOHK
+-- License: MIT
+--
+-- | Module provides wallet layer api that is used by API layer
+-- | and uses both DB and Networking layer to realize its role
+-- | as being intermediary between the three.
+
+
+module Cardano.WalletLayer where
+
+import Cardano.Wallet
+    ( Wallet )
+import Cardano.WalletLayer.Wallet
+    ( CreateWallet, WalletId )
+import Control.Monad.Except
+    ( ExceptT )
+
+data WalletLayerError
+
+data WalletLayer m s = WalletLayer
+    { createWallet :: CreateWallet -> ExceptT WalletLayerError m (Wallet s)
+    , getWallet :: WalletId -> ExceptT WalletLayerError m (Wallet s)
+    }
```

Comment:
Let's have separate error types for the different functions. I am not expecting `getWallet` to throw anything related to the creation of a wallet.

Reply:
done